internal/operator-metrics/metrics.go

```diff
@@ -0,0 +1,142 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operatormetrics
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	// namespaceFile is the path to the namespace file for the service account.
+	namespaceFile = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+
+	// caBundleConfigMap declares the name of the config map for the CA bundle.
+	caBundleConfigMap string = "serving-certs-ca-bundle"
+
+	// prometheusCAFile declares the path for prometheus CA file for service monitors in OpenShift.
+	prometheusCAFile string = fmt.Sprintf("/etc/prometheus/configmaps/%s/service-ca.crt", caBundleConfigMap)
+
+	// nolint #nosec
+	// bearerTokenFile declares the path for bearer token file for service monitors.
+	bearerTokenFile string = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+
+	// openshiftInClusterMonitoringNamespace declares the namespace for the OpenShift in-cluster monitoring.
+	openshiftInClusterMonitoringNamespace string = "openshift-monitoring"
+)
+
+type OperatorMetrics struct {
+	kubeClient client.Client
+}
+
+func NewOperatorMetrics(config *rest.Config, scheme *runtime.Scheme) (OperatorMetrics, error) {
+	kubeClient, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return OperatorMetrics{}, err
+	}
+
+	return OperatorMetrics{
+		kubeClient: kubeClient,
+	}, nil
+}
+
+func (d OperatorMetrics) Start(ctx context.Context) error {
+	rawNamespace, err := os.ReadFile(namespaceFile)
+	if err != nil {
+		return fmt.Errorf("error reading namespace file: %w", err)
+	}
+	namespace := string(rawNamespace)
+
+	var tlsConfig *monitoringv1.TLSConfig
+
+	if d.caConfigMapExists() {
+		serviceName := fmt.Sprintf("opentelemetry-operator-controller-manager-metrics-service.%s.svc", namespace)
+
+		tlsConfig = &monitoringv1.TLSConfig{
+			CAFile: prometheusCAFile,
+			SafeTLSConfig: monitoringv1.SafeTLSConfig{
+				ServerName: &serviceName,
+			},
+		}
+	} else {
+		t := true
+		tlsConfig = &monitoringv1.TLSConfig{
+			SafeTLSConfig: monitoringv1.SafeTLSConfig{
+				// kube-rbac-proxy uses a self-signed cert by default
+				InsecureSkipVerify: &t,
```

Comment:
I think we talked about this internally as a bug that we should not be using the `InsecureSkipVerify` it is also not present here https://github.com/os-observability/konflux-opentelemetry/blob/main/bundle-patch/manifests/opentelemetry-operator-metrics-monitor_monitoring.coreos.com_v1_servicemonitor.yaml#L22

Reply:
If you review the code, you will see this will be only executed if the ConfigMap containing the CA doesn´t exist. This cover non-OpenShift scenarios.